File: Apps4Bro.iOS/Networks/DummyNetwork.cs
using System;

namespace Apps4Bro.Networks
{
    internal class DummyNetwork : AdNetworkHandler
    {
        private readonly AdManager m_adManager;
        private string m_id;
        private object m_data;

        public string Network
        {
            get { return "Dummy"; }
        }

        public DummyNetwork(AdManager manager)
        {
            m_adManager = manager;
        }

        public bool Show(object data)
        {
            m_data = data;

            return true;
        }

        public void Hide()
        {
            
        }

        public void Display()
        {
            
        }

        public void SetId(string id)
        {
            m_id = id;
        }

    }
}

File: Apps4Bro.Swift/Sources/Apps4Bro/Networks/DummyNetwork.swift
import Foundation

// Port of DummyNetwork.cs. No-op stub useful for routing tests through AdManager
// without hitting a real ad network.
final class DummyNetwork: AdNetworkHandler {
    private weak var manager: AdManager?
    private var id: String = ""

    let network: String = "Dummy"

    init(manager: AdManager) {
        self.manager = manager
    }

    func show(data: Any) -> Bool { true }
    func hide() {}
    func display() {}
    func setId(_ id: String) { self.id = id }
}
